### ISSUE TITLE
Z-wave inclusion doesn't work - stuck at Exchanging security key and other issues

### ISSUE THREAD

**Issue by djanssen1** [athombv/homey#1678]
**What is your Homey version (Settings → System)?**

1.3.2

**What did you try to achieve?**

Include a Philio device

*Please include detailed steps how we can reproduce the issue*

Tried 3 times to include and everytime it gets stuck at the phase it showing "exchanging security key" after a while (timeout?) it asked again to put it into inclusion mode as it is paired, it gives error that it is already paired. Then first delete it through Z-wave settings and tried to include again. After 3 times I decided to reboot and then it worked directly. I have seen this behaviour on 1.3 RCs as well and then also reboot fixed the issue.
Errors like Error: queue_full are visible in the Z-wave log

Then performed reboot. Then tried again, it got further when retrieving classes it stopped and showing that it could not reach despite the fact the device is 5 cm from Homey
After Homey seems to be very busy. When trying to add a device again it it slower and when Z-wave device creation starts it shows for a long period of time waiting z-wave to be ready
Then I had to remove node again (twice... ) and after that it was succesfull included

**Please attach a Homey System log (Settings -> System -> Send Log)**

After 3 times and before reboot: FDA8C0F1E2
After reboot, other issues and succesful inclusion: 308CE686C3

...

**Comment by selzilla** [comment]
The same problem. Affected Fibaro Wall Plugs.
Can not be integrated without restarting Homey.
Then the Wallplugs are available for 5 minutes before setting the work.
They are still visible, but they are no longer responsive to any circuits.
Wallplugs that have been plugged in a while can work without problems.

**Comment by djanssen1** [comment]
HI @selzilla In my opinion this seems to be not same problem. Anyway it is very difficult for you to judge... What I see is different as you indicate your devices are not working. Also your English is difficult to understand (have you translated it?) Maybe better to report in Dutch to Athom support. Also if you report an issue you should use the template and good description what you exactly have done and include logs...

**Comment by djanssen1** [comment]
Today with next inclusion of a device same issue. One z-wave device (Aeotec ZW089) has been broken. So first try to remove it but that didn't work. Tried 3 times, then gave up.
Then started to include a new one...

3 times trying. First time it gave error that it wasn't able to reach device anymore (device is next to Homey) other 2 times same issue, stuck at Exchanging security key. After a while (timeout?) it asked for pressing inclusion button and then it told that it was already paired. So first had to manually unpair it through z-wave settings.

After PTP first tried to remove broken device; that worked directly
Also inclusion of new device was succesful directly first time

Logs before PTP of Homey: C3BD66357E
Logs after PTP and succesful inclusion: DD8D4118C9

**Comment by djanssen1** [comment]
And again same issue with another new device I'm trying to include. Also stuck at Exchanging security key. Tried 3 times...

After PTP it was able to succesful include

Log: C52697F58C

**Comment by jeroenvollenbrock** [comment]
Your log shows a Z-Wave device (node 90) that is flooding the Z-Wave network with traffic, which is disrupting the communication with other devices. Maybe the reporting interval is too short?

**Comment by Sillec123** [comment]
i got the same issue

**Comment by djanssen1** [comment]
I'm now on 1.5 RC2 and again doing some exclusion/inclusion of new devices and have same behaviour. @jeroenvollenbrock indicates that Node90 was flooding the network. I did some check and couldn't find that much of traffic. Again here it is stuck at exchanging security. Logs before reboot 20B37C526A after reboot things went smoothly.

**Comment by djanssen1** [comment]
Updated to 1.5 RC4 and after a minute or 10 added two new z-wave devices and both where added in one go. After last device has added I got the message Homey Offline once. Based on Slack communication it is due to crashing app. Have no clue what app was crashed. Device I added was a Philio device. Philio app is from app store. Logs: 5BED605DC8

**Comment by djanssen1** [comment]
RC5; still same issue exists. Trying to add a Fibaro Dimmer 2; stuck at exchanging key

Log: E82C04079C

Part of log

[2017-08-10T20:21:12.288Z] Command[6131] status: addNode, status: ADD_NODE_STATUS_NODE_FOUND
[2017-08-10T20:21:12.870Z] Command[6131] status: addNode, status: ADD_NODE_STATUS_ADDING_SLAVE
[2017-08-10T20:21:17.388Z] Command[6131] status: addNode, status: ADD_NODE_STATUS_PROTOCOL_DONE
[2017-08-10T20:21:17.447Z] Command[6131] status: addNode, status: ADD_NODE_STATUS_DONE
[2017-08-10T20:21:18.418Z] Command[6131] status: addNode, status: INTERVIEW_SECURITY
[2017-08-10T20:21:18.420Z] Node[184]: sendData to COMMAND_CLASS_SECURITY, params 0x0400
[2017-08-10T20:21:18.422Z] Command[6132] start: sendData
[2017-08-10T20:21:18.424Z] ProcessSendData[6128]: To node: 184 with data: 0x980400 and txOptions: ACK,AUTO_ROUTE,EXPLORE
[2017-08-10T20:21:18.601Z] Command[6132] end: sendData
[2017-08-10T20:21:18.629Z] Node[184]: Received application command for COMMAND_CLASS_SECURITY, data: 0x0500

**Comment by djanssen1** [comment]
Running 1.5 RC5
Tried after a day again to add Fibaro Dimmer 2; same situation. Stuck at Exchanging security key
Log: 92B1F34906

Then rebooted Homey and after a minute or 10 tried to do it again. Still taking quite some time for exchanging security key but then it successfully included the device. After I got the message it was successful included I got the Homey Off-line message but device was nicely created.

Log after success: 0FE8EEB4CA
